compiler/rustc_resolve/src/imports.rs

```diff
@@ -1094,7 +1117,7 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
             redundant_spans.dedup();
             self.r.lint_buffer.buffer_lint_with_diagnostic(
                 UNUSED_IMPORTS,
-                import.id,
+                import.expect_id(),
```

Comment:
Why can we `expect` the id?  Could you add a comment+assert on `check_for_redundant_imports` stating that is must only be called for `ImportKind::Single`?